pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -377,6 +385,13 @@ func (rd *restoreDataProcessor) processRestoreSpanEntry(
 	iter := sst.iter
 	defer sst.cleanup()
 
+	writeAtNow := restoreAtNow.Get(&evalCtx.Settings.SV)
+	if writeAtNow && !evalCtx.Settings.Version.IsActive(ctx, clusterversion.MVCCAddSSTable) {
+		return roachpb.BulkOpSummary{}, errors.Newf(
+			"cannot use %s until version %s", restoreAtNow.Key(), clusterversion.MVCCAddSSTable.String(),
+		)
+	}
```

Comment:
Is the plan to enable this setting by default for 22.1. If so, won't we have to silently fall back to the old behaviour in those cases?

If we're planning on keeping this check, it might be better to add a custom validation function for the setting that rejects enabling it if the version gate is not enabled.

Reply:
The plan is to remove this setting before 22.1 I think, and move to a pair of cluster version gates.